common/models/user.js

```diff
@@ -301,6 +301,19 @@ module.exports = function(User) {
     return fn.promise;
   };
 
+  User.deleteById = function(tokenId, fn) {
```

Comment:
This is weird. You are overriding built-in `PersistedModel.deleteById` implementation provided by jugglers' DataAccessObject, and change the semantics so that it no longer deletes the User instance, but deletes an access token only. That does not look right to me.

I think the correct solution is to use a "before delete" hook and delete *all* access tokens of the user from the database.